ni_measurement_plugin_sdk_service/_internal/parameter/encoder.py

```diff
@@ -0,0 +1,206 @@
+# enums and default values
+from enum import Enum
+from typing import Any, Dict, Sequence
+
+from google.protobuf import descriptor_pb2, descriptor_pool, message_factory
+from google.protobuf.descriptor_pb2 import FieldDescriptorProto
+
+from ni_measurement_plugin_sdk_service._internal.parameter.decoder_strategy import (
+    get_type_default,
+)
+from ni_measurement_plugin_sdk_service._internal.parameter.metadata import (
+    ParameterMetadata,
+)
+from ni_measurement_plugin_sdk_service.measurement.info import ServiceInfo
+from tests.utilities.stubs.loopback.types_pb2 import ProtobufColor
+
+
+def serialize_parameters(
+    parameter_metadata_dict: Dict[int, ParameterMetadata],
+    parameter_values: Sequence[Any],
+    service_info: ServiceInfo,
+) -> bytes:
+    """Serialize the parameter values in same order based on the metadata_dict.
+
+    Args:
+        parameter_metadata_dict (Dict[int, ParameterMetadata]): Parameter metadata by ID.
+
+        parameter_value (Sequence[Any]): Parameter values to serialize.
+
+        Service_info (ServiceInfo): Unique service name.
+
+    Returns:
+        bytes: Serialized byte string containing parameter values.
+    """
+    pool = descriptor_pool.Default()
+    message_name = "".join(char for char in service_info.service_class if char.isalpha())
+    # Tries to find a message type in pool with message_name else it creates one
+    try:
+        message_proto = pool.FindMessageTypeByName(f"{message_name}.{message_name}")
+    except KeyError:
+        message_proto = _create_message_type(
+            parameter_values, parameter_metadata_dict, message_name, pool
+        )
+    message_instance = message_factory.GetMessageClass(message_proto)()
+
+    for i, parameter in enumerate(parameter_values, start=1):
+        field_name = f"field_{i}"
```

Comment:
I would prefer to use the configuration/output names specified by the measurement service. Using numeric field names loses some of the debugging benefits of protobuf, like the ability to print a message as a string. 

You may need to "sanitize" them, if protobuf won't accept spaces or other weird characters that we allow. If you do this, it may make sense for the `ParameterMetadata` to save the sanitized name so we don't have to build a bunch of temporary strings whenever we serialize fields.

Reply:
Protobuf field names should be in snake_case, but ParameterMetadata.sanitized_display_name is removing the underscores.

I was suggesting that you add the sanitized name to ParameterMetadata instead of building a new string every you serialize or deserialize a property. Does this matter for performance? I don't know, I'd have to benchmark it. A compromise would be to scan the string for disallowed characters and only build a new string when any are present.